test/lib/recordevents/event_info_store.go

```diff
@@ -246,7 +246,7 @@ func (ei *EventInfoStore) AssertAtLeast(min int, matchers ...EventInfoMatcher) [
 // This method fails the test if the assert is not fulfilled.
 func (ei *EventInfoStore) AssertInRange(min int, max int, matchers ...EventInfoMatcher) []EventInfo {
 	events := ei.AssertAtLeast(min, matchers...)
-	if max > 0 && len(events) > max {
+	if len(events) > max {
```

Comment:
the logic is incorrect. you might have gotten `<= max` events but this is not testing that those events that are counted match the expected matchers. So you could be over the max event count but under on matched event counts.